## Data Wrangling
In this notebook I'll be demonstrating a few steps and procedures one might take in cleaning up a messy dataset. Most data encountered in the wild will require some (or a lot) of cleaning and processing. The currated datasets you may encounter in other tutorials being the exception. 

In [26]:
import pandas as pd
import numpy as np

In [4]:
df = pd.read_csv('Health Data.csv')

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 865 entries, 0 to 864
Data columns (total 7 columns):
Start                     865 non-null object
Finish                    865 non-null object
Active Calories (kcal)    865 non-null float64
Dietary Calories (cal)    865 non-null float64
Distance (mi)             865 non-null float64
Steps (count)             865 non-null float64
Weight (lb)               865 non-null float64
dtypes: float64(5), object(2)
memory usage: 47.4+ KB


In [15]:
df.columns = ['start', 'finish', 'active', 'diet', 'distance', 'steps', 'weight']
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 865 entries, 0 to 864
Data columns (total 7 columns):
start       865 non-null object
finish      865 non-null object
active      865 non-null float64
diet        865 non-null float64
distance    865 non-null float64
steps       865 non-null float64
weight      865 non-null float64
dtypes: float64(5), object(2)
memory usage: 47.4+ KB


There's no way I actually completely logged 865 days of calories and weight 

In [22]:
logged_diet = df[df['diet'] > 0]
logged_weight = df[df['weight'] > 0]
logged_diet_weight = logged_diet[logged_diet['weight'] > 0]

print('info on days that diet was logged')
print('-----' * 20)
print(logged_diet.info())
print()
print('info on days that weight was logged')
print('-----' * 20)
print(logged_weight.info())
print()
print('info on days that diet and weight was logged')
print('-----' * 20)
print(logged_diet_weight.info())

info on days that diet was logged
----------------------------------------------------------------------------------------------------
<class 'pandas.core.frame.DataFrame'>
Int64Index: 29 entries, 520 to 558
Data columns (total 7 columns):
start       29 non-null object
finish      29 non-null object
active      29 non-null float64
diet        29 non-null float64
distance    29 non-null float64
steps       29 non-null float64
weight      29 non-null float64
dtypes: float64(5), object(2)
memory usage: 1.8+ KB
None

info on days that weight was logged
----------------------------------------------------------------------------------------------------
<class 'pandas.core.frame.DataFrame'>
Int64Index: 28 entries, 521 to 857
Data columns (total 7 columns):
start       28 non-null object
finish      28 non-null object
active      28 non-null float64
diet        28 non-null float64
distance    28 non-null float64
steps       28 non-null float64
weight      28 non-null float64
dtypes: float64(

In [23]:
np.beta(,)

In [33]:
logged_diet.diet.quantile(.5)

1179190.0